File: csharp/Production/cube/RubikFileReader.cs
using System;

using System.IO;

namespace cube
{
    public class RubikFileReader
    {
        Boolean c_fileIsOK;

        StreamReader   c_fileReader;

        public RubikFileReader()
        {
        }

        public RubikFileReader(String p_fileLocation)
        {
            try
            {
                c_fileReader = new StreamReader(p_fileLocation);
                c_fileIsOK = true;
            }
            catch (IOException ex)
            {
                Console.WriteLine(ex.Message);
                c_fileIsOK = false;
            }
        }

        public int read()
        {
            if (!c_fileIsOK)
                return -1;
            else
                try
                {
                    int readChar = c_fileReader.Read();
                    return readChar;
                }
                catch (IOException ex)
                {
                    Console.WriteLine(ex);
                    return -1;
                }
        }

        public void close()
        {
            try
            {
                c_fileReader.Close();
            }
            catch (IOException ex) { Console.WriteLine(ex); }
        }
    }
}

File: python/production/cube/rubik_file_reader.py
class Rubik_file_reader:

        def __init__(self, p_fileLocation):
            self.c_fileReader = open(p_fileLocation,"r")
            self.c_fileIsOK = True

        def read(self):
            if not self.c_fileIsOK:
                return -1
            else:
                readChar = self.c_fileReader.read(1)
                return readChar

        def close(self):
            self.c_fileReader.close()
